### PR TITLE
Issue#31 functionality in search walker

### LINKED ISSUE #31: Functionality in Search Walker
# REQUIREMENTS
Set the functionality when clicking the Search Walker button on the Home screen

# DESIGN

# Acceptance Criteria
List all required acceptance criteria
- [ ] Button functionality
- [ ] Set Navigation
- [ ] Show walkers
- [ ] Adapt it to the attached design

### PR DESCRIPTION
**Pull request by Darkboy5120**
# Description
the focus in this pull was introduce the first version of walkers and clients positions match, the algoritm check boths users's distance to show to the client nears walkers, so boths users are updating theirs position to server in order to find near users.

### CODE REVIEW

File: src/assets/controllers/firebaseShortcuts.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  return firebase.firestore().collection(collection);
+};
+
+const getDoc = (collection, document) => {
+  return getCollection(collection).doc(document);
+}
+
+const handleThen = (req, callback) => {
+  return req.catch(e => {
+    console.error(e);
```
Comment (YilKanda) [inline src/assets/controllers/firebaseShortcuts.js]: Don't put console.log(), when you push your code from GitHub. Instead you can make use of Alert.alert()

File: src/assets/controllers/firebaseShortcuts.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+const fbShortcuts = {
+  getUserByUID,
+  updateDoc,
+  add,
+  getCollection,
+};
+
+export default fbShortcuts;
```
Comment (YilKanda) [inline src/assets/controllers/firebaseShortcuts.js]: Add a blank line

File: src/assets/controllers/updateRealtimeLocation.js [hunk 3]
```diff
@@ -71,6 +68,7 @@ const listen = async () => {
 
 const realtimeLocation = {
   listen: listen,
+  // listen: () => {},
```
Comment (YilKanda) [inline src/assets/controllers/updateRealtimeLocation.js]: Don't add code in comments on GitHub.

File: src/components/LoadingSpinner/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  return result;
+};
+
+const LoadingSpinner = ({size, scale}) => {
+  return (
+    <ActivityIndicator size={parseInt(getSize(size))*(parseInt(scale) ?? 1)} color={theme.color.primary1} />
+  );
+};
+
+export default LoadingSpinner;
```
Comment (YilKanda) [inline src/components/LoadingSpinner/index.js]: Add a blank line

**Review by LuisGomez89 — APPROVED** [review]

File: src/assets/controllers/firebaseShortcuts.js (hunk 1 above)
Comment (Darkboy5120) [reply src/assets/controllers/firebaseShortcuts.js]: done

File: src/assets/controllers/firebaseShortcuts.js (hunk 2 above)
Comment (Darkboy5120) [reply src/assets/controllers/firebaseShortcuts.js]: done

File: src/components/LoadingSpinner/index.js (hunk 4 above)
Comment (Darkboy5120) [reply src/components/LoadingSpinner/index.js]: done

File: src/assets/controllers/updateRealtimeLocation.js (hunk 3 above)
Comment (Darkboy5120) [reply src/assets/controllers/updateRealtimeLocation.js]: i forgot remove it

**Review by YilKanda — APPROVED** [review]
I recommend that when you make fixes in the code, instead of creating a commit for each fix, you can use the Git amend:

```
git add .
git commit --amend --no-edit
git push --force-with-lease origin your-branch
```

**Review by adrianagh13 — APPROVED** [review]

**Pull request merged**